util/src/main/java/io/kubernetes/client/util/authenticators/GCPAuthenticator.java

```diff
@@ -58,6 +70,27 @@ public boolean isExpired(Map<String, Object> config) {
 
   @Override
   public Map<String, Object> refresh(Map<String, Object> config) {
-    throw new IllegalStateException("Unimplemented");
+    if (!config.containsKey(CMD_ARGS) || !config.containsKey(CMD_PATH))
+      throw new RuntimeException("Could not refresh token");
+    String cmd_path= (String) config.get(CMD_PATH);
+    String cmd_args = (String) config.get(CMD_ARGS);
+    List<String> commandList = new ArrayList<>(Arrays.asList(cmd_args.split(" ")));
+    commandList.add(0, cmd_path);
+    try {
+      Process process = new ProcessBuilder().command(commandList).start();
+      process.waitFor(10, TimeUnit.SECONDS);
+      if(process.exitValue() != 0) {
+        throw new IOException("Process exit code: "+process.exitValue());
```

Comment:
I'm not sure that IOException is right here. `IllegalStateException(...)`?

Reply:
Also, please make the message more descriptive and include the stderr to make the exception more debuggable.

`throw new IllegalStateException("Failed to get token (" + process.exitValue() + ") " + stderr`